ironfish/src/syncer.ts

```diff
@@ -358,23 +359,34 @@ export class Syncer {
         )} (${sequence}) from ${peer.displayName}`,
       )
 
-      const start = BenchUtils.start()
-
-      const [headBlock, ...blocks]: SerializedBlock[] = await this.peerNetwork.getBlocks(
-        peer,
-        head,
-        this.blocksPerMessage + 1,
-      )
-
-      const elapsedSeconds = BenchUtils.end(start) / 1000
-      this.downloadSpeed.add((blocks.length + 1) / elapsedSeconds)
+      const blocksResponse = await blocksPromise
+      if (blocksResponse instanceof Error) {
+        throw blocksResponse
+      }
```

Comment:
I'm not totally sure how this works -- it'd be nice if we could find a different way to solve the issue of logging an error when closing the node, or if not, I think we should add a comment explaining this.